src/tracing/SwapEventTimestampTracker.ts

```diff
@@ -0,0 +1,51 @@
+import { calculateElapsedTimeWithPerformanceMark } from './utils'
+
+export enum SwapEventType {
+  FIRST_SWAP_ACTION = 'FIRST_SWAP_ACTION',
+  FIRST_QUOTE_FETCH_STARTED = 'FIRST_QUOTE_FETCH_STARTED',
+  FIRST_SWAP_SIGNATURE_REQUESTED = 'FIRST_SWAP_SIGNATURE_REQUESTED',
+  FIRST_SWAP_SIGNATURE_COMPLETED = 'FIRST_SWAP_SIGNATURE_COMPLETED',
+  FIRST_SWAP = 'FIRST_SWAP',
```

Comment:
nit: `FIRST_SWAP_SUCCESS` is more direct and would match the helper fn `logSwapSuccess`